server/src/ankaios_server.rs

```diff
@@ -1643,4 +1654,37 @@ mod tests {
 
         assert!(comm_middle_ware_receiver.try_recv().is_err());
     }
+
+    // [utest->swdd~server-receives-resource-availability~1]
+    #[tokio::test]
+    async fn utest_server_recieves_agent_resource_availability_info() {
+        let payload = AgentLoadStatus {
+            agent_name: AGENT_A.to_string(),
+            agent_resources: AgentLoad {
+                cpu_usage: 42,
+                free_memory: 42,
+            },
+        };
+
+        let _ = env_logger::builder().is_test(true).try_init();
+        let (to_server, server_receiver) = create_to_server_channel(common::CHANNEL_CAPACITY);
+        let (to_agents, _comm_middle_ware_receiver) =
+            create_from_server_channel(common::CHANNEL_CAPACITY);
+
+        let mut server = AnkaiosServer::new(server_receiver, to_agents);
+        let mut mock_server_state = MockServerState::new();
+        mock_server_state
+            .expect_update_agent_resource_availability()
+            .with(mockall::predicate::eq(payload.clone()))
+            .return_const(());
+        server.server_state = mock_server_state;
+
+        let agent_resource_result = to_server.agent_load_status(payload).await;
+        assert!(agent_resource_result.is_ok());
+
+        drop(to_server);
+        let result = server.start(None).await;
+
+        assert!(result.is_ok());
```

Comment:
This is too weak, can you assert that you receive exactly the message you are expexting.

Reply:
Solved as per discussion.

Reply:
How was this solved?